### PR TITLE
Fixes #26352 - ActiveJob task arguments loading

### PR DESCRIPTION
**Pull request by ezr-ondrej**
ActiveJob arguments loading is broken by: https://github.com/Dynflow/dynflow/pull/317

### CODE REVIEW

File: app/models/foreman_tasks/task/dynflow_task.rb [hunk 1]
```diff
@@ -118,7 +118,8 @@ def main_action
                else
                  execution_plan_action.input
                end
-        @main_action = active_job_action(args['job_class'], args['job_arguments'])
+        args = args['job_data'] || args.slice('job_class')
```
Comment (iNecas) [inline app/models/foreman_tasks/task/dynflow_task.rb]: For the tasks created before this change, shouldn't this be:

```ruby
args = args['job_data'].try { |data| data['arguments'] } || args['job_arguments']
@main_action = active_job_action(args['job_class'], args['arguments'])
```

Comment (ezr-ondrej) [reply app/models/foreman_tasks/task/dynflow_task.rb]: Yes, you are right :+1: thanks.

**Comment by ezr-ondrej** [comment]
I believe now we are backwards compatible :)

File: app/models/foreman_tasks/task/dynflow_task.rb (hunk 1 above)
Comment (iNecas) [reply app/models/foreman_tasks/task/dynflow_task.rb]: One small ask: please add a comment that we need the fallback for `job_arguments` for fobs from Dynflow `<=1.2.1`, for the future version of us that would wonder why we're doing this here :)

**Comment by iNecas** [comment]
There is one more occurrence of `job_arguments` in https://github.com/theforeman/foreman-tasks/blob/1c853d37cd1a7a66ea07b57280046070dbcc5041/app/models/foreman_tasks/task/dynflow_task.rb#L86, would be good to fix that one as well while touching it.

**Comment by ezr-ondrej** [comment]
I have pulled the logic into its method, just left one comment, where I am unsure.

**Comment by iNecas** [comment]
Thanks @ezr-ondrej

**Pull request merged**